shell/platform/windows/direct_manipulation.cc

```diff
@@ -53,16 +56,31 @@ HRESULT DirectManipulationEventHandler::OnViewportStatusChanged(
             (int32_t) reinterpret_cast<int64_t>(this));
       }
     }
-  } else if (previous == DIRECTMANIPULATION_RUNNING) {
+  }
+  if (previous == DIRECTMANIPULATION_RUNNING) {
+    // Reset deltas to ensure only inertia values will be compared later.
+    last_pan_delta_x_ = 0.0;
+    last_pan_delta_y_ = 0.0;
     if (owner_->binding_handler_delegate) {
       owner_->binding_handler_delegate->OnPointerPanZoomEnd(
           (int32_t) reinterpret_cast<int64_t>(this));
     }
+  } else if (previous == DIRECTMANIPULATION_INERTIA) {
+    if (owner_->binding_handler_delegate &&
+        std::max(std::abs(last_pan_delta_x_), std::abs(last_pan_delta_y_)) >
+            0.01) {
+      owner_->binding_handler_delegate->OnScrollInertiaCancel(
+          (int32_t) reinterpret_cast<int64_t>(this));
```

Comment:
Can you extract `(int32_t) reinterpret_cast<int64_t>(this)` into a separate function such as `deviceId` with a short comment? This statement seems fishy and appears several times in this file.